Lib/Protocols/IdFTPServer.pas

```diff
@@ -7373,4 +7562,145 @@ procedure TIdFTPServerContext.ResetZLibSettings;
   FZLibStratagy := DEF_ZLIB_STRATAGY; // - default
 end;
 
+{ TIdFTPServerInfo }
+
+constructor TIdFTPServerInfo.Create;
+begin
+  inherited Create;
+  FAdditionalFacts := TStringList.Create;
+{$IFDEF HAS_TStringList_CaseSensitive}
+  TStringList(FAdditionalFacts).CaseSensitive := False;
+{$ENDIF}
+end;
+
+destructor TIdFTPServerInfo.Destroy;
+begin
+  FreeAndNil( FAdditionalFacts);
+  inherited Destroy;
+end;
+
+procedure TIdFTPServerInfo.SetAdditionalFacts(const AValue: TStrings);
+begin
+  FAdditionalFacts.Assign(AValue);
+end;
+
+{ TIdFTPClientInfo }
+
+constructor TIdFTPClientInfo.Create;
+begin
+  inherited Create;
+  FFacts := TStringList.Create;
+{$IFDEF HAS_TStringList_CaseSensitive}
+  TStringList(FFacts).CaseSensitive := False;
+{$ENDIF}
+end;
+
+destructor TIdFTPClientInfo.Destroy;
+begin
+  FreeAndNil(FFacts);
+  inherited;
+end;
+
+function TIdFTPClientInfo.GetClientName: String;
+begin
+  Result := FFacts.Values['Name'];
+end;
+
+function TIdFTPClientInfo.GetClientVersion: String;
+begin
+  Result := FFacts.Values['Version'];
+end;
+
+function TIdFTPClientInfo.GetCLNTParams: String;
+begin
+  if ClientName <> '' then
+  begin
+    Result := ClientName;
+    if ClientVersion <> '' then
+    begin
+      Result := Result + ' ' + ClientVersion;
+      if PlatformDescription <> '' then
+      begin
+        Result := Result + ' '+ PlatformDescription;
+      end;
+    end;
+  end
+  else
+  begin
+    Result := '';
+  end;
+end;
+
+function TIdFTPClientInfo.GetCSIDParams: String;
+var i : Integer;
+begin
+  Result := '';
+  for i := 0 to FFacts.Count - 1 do
+  begin
+    Result := '; '+FFacts[i];
+  end;
+  IdDelete(Result,1,2);
+end;
+
+function TIdFTPClientInfo.GetPlatformDescription: String;
+begin
+  Result := FFacts.Values['OS'];
+end;
+
+function TIdFTPClientInfo.GetPlatformVersion: String;
+begin
+  Result := FFacts.Values['OSVer'];
+end;
+
+function TIdFTPClientInfo.GetVendor: String;
+begin
+  Result := FFacts.Values['Vendor'];
+end;
+
+procedure TIdFTPClientInfo.SetClientName(const AValue: String);
+begin
+  FFacts.Values['Name'] := AValue;
+end;
+
+procedure TIdFTPClientInfo.SetClientVersion(const AValue: String);
+begin
+  FFacts.Values['Version'] := AValue;
+end;
+
+procedure TIdFTPClientInfo.SetCLNTParams(const AValue: String);
+var LBuf : String;
+begin
+  LBuf := TrimLeft(Avalue);
+  ClientName := Fetch(LBuf);
+  ClientVersion := Fetch(LBuf);
+  PlatformDescription := LBuf;
+end;
+
+procedure TIdFTPClientInfo.SetCSIDParams(const AValue: String);
+var LBuf,
+    LValue : String;
+begin
+   FFacts.Clear;
+   LBuf := AValue;
+   repeat
+      LValue := TrimLeft(Fetch(LBuf,';'));

```

Comment:
Perhaps replace this whole loop with a single call to `IdGlobal.SplitDelimitedString()` instead?

`SplitDelimitedString(AValue, FFacts, True, ';');`